src/Raven.Server/Utils/TypeConverter.cs

```diff
@@ -544,8 +544,8 @@ public static dynamic ToDynamicType(object value)
             return value;
         }
 
-        private static readonly List<string> _timeSpanPropertiesNames = typeof(TimeSpan).GetProperties().Select(i => i.Name).ToList();
-        private static readonly List<string> _timeOnlyPropertiesNames = typeof(TimeOnly).GetProperties().Select(i => i.Name).ToList();
+        private static readonly HashSet<string> _timeSpanPropertiesNames = typeof(TimeSpan).GetProperties().Select(i => i.Name).ToHashSet();
+        private static readonly HashSet<string> _timeOnlyPropertiesNames = typeof(TimeOnly).GetProperties().Select(i => i.Name).ToHashSet();
```

Comment:
We do below `.Contains()` so switching to `HashSet` should be slightly better perf

https://github.com/ravendb/ravendb/pull/17716/files#diff-6a3d3e756a23fd2123c4d5c970e0baa03f3ebf895169602efdef19edcc7f1461R565